src/annotator/components/NotebookModal.js

```diff
@@ -0,0 +1,100 @@
+import { useEffect, useRef, useState } from 'preact/hooks';
+import classnames from 'classnames';
+
+import { createSidebarConfig } from '../config/sidebar';
+
+// FIXME: use the button from the frontend shared package once this is stable.
+import Button from '../../sidebar/components/Button';
+
+/**
+ *
+ * @typedef NotebookIframeProps
+ * @prop {Record<string, any>} config
+ * @prop {string} groupId
+ */
+
+/**
+ * Create the iframe that will load the notebook application.
+ * 
+ * @param {NotebookIframeProps} props
+
+ */
+function NotebookIframe({ config, groupId }) {
+  const notebookConfig = createSidebarConfig(config);
+  // Explicity set the "focused" group
+  notebookConfig.group = groupId;
+  const configParam = encodeURIComponent(JSON.stringify(notebookConfig));
+  const notebookAppSrc = `${config.notebookAppUrl}#config=${configParam}`;
+
+  return (
+    <iframe
+      key={groupId} // force re-rendering on group change
+      title={'Hypothesis annotation notebook'}
+      className="Notebook__iframe"
+      // Enable media in annotations to be shown fullscreen
+      allowFullScreen
+      src={notebookAppSrc}
+    />
+  );
+}
+
+/**
+ * @typedef NotebookModalProps
+ * @prop {import('../util/emitter').EventBus} eventBus
+ * @prop {Record<string, any>} config
+ */
+
+/**
+ *
```

Comment:
Can you add a brief comment here to explain what this component is. That's obvious to all of us working on it now, but it will be helpful for new developers in future.

Reply:
Sure